presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/IterativeOptimizer.java

```diff
@@ -106,6 +105,7 @@ private boolean exploreGroup(int group, Context context)
     private boolean exploreNode(int group, Context context)
     {
         PlanNode node = context.getMemo().getNode(group);
+        PlanNodeMatcher planNodeMatcher = new PlanNodeMatcher(context.getLookup());
```

Comment:
Done. Passing separately, as plan node matcher should not leak to Rule-s via Rule.Context.